### ISSUE TITLE
React Select `onChange` should match signature of other form fields

### ISSUE THREAD

**Issue by zboman** [thumbtack/thumbprint#188]
Currently the `onChange` handler for `Input` looks like this:

```js
<Input onChange={(value, event) => {
    console.log(value, event); // "text" SyntheticEvent{}
}} />
```

But the `onChange` for `Select` only receives the `value` and never receives the event:

```js
<Select onChange={(value, event) => {
    console.log(value, event); // "text" undefined
}} />
```

I'd like to propose we modify the signature of `Select`'s `onChange` to match that of `Input`.

**Comment by danoc** [comment]
Thanks for the issue, @zboman!

Out of curiosity, what's the use case you ran into where this is helpful?

I'd be happy to accept a PR for this one.

**Comment by zboman** [comment]
We've got a `<form>` of fields and the `form` has a single `onFieldChange` handler that we'd like to pass to all of the field elements to track changes. One of the things we need from each of the fields is the `event.target.name` attribute off of the element in order to uniquely identify it. This works great for `<Input>` but obviously fails for `<Select>`.

The handler looks roughly like this (though I've simplified it some):
```js
onFieldChange = (value, e) => {
    this.setState({ [e.target.name]: value });
}
```
